app/api/domains/osu.py

```diff
@@ -484,48 +469,28 @@ async def osuSearchHandler(
         # convert to osu!api status
         params["status"] = RankedStatus.from_osudirect(ranked_status).osu_api
 
-    if USING_NASUYA:
-        # nasuya can serialize to direct for us
-        params["osu_direct"] = True
-
-    async with app.state.services.http_client.get(search_url, params=params) as resp:
+    async with app.state.services.http_client.get(
+        app.settings.MIRROR_SEARCH_ENDPOINT,
+        params=params,
+    ) as resp:
         if resp.status != status.HTTP_200_OK:
-            if USING_CHIMU:
-                # chimu uses 404 for no maps found
-                if resp.status == status.HTTP_404_NOT_FOUND:
-                    return b"0"
-
             return b"-1\nFailed to retrieve data from the beatmap mirror."
 
-        if USING_NASUYA:
-            # nasuya returns in osu!direct format
-            return await resp.read()
```

Comment:
should still keep this case as it's a significant simplification

Reply:
hmm I don't know, I don't think it has any benefits. The time of quickly converting it is like close to nothing and because of that the extra code is just unnecessary. Just have mirrors use cheesegull and its all good

Reply:
cheesegull is really not a great protocol lol

Reply:
But if you really plead to, maybe we should instead make a list of "direct direct" mirrors OR what would be better pass a query parameter to the mirror that the mirror has to handle which would be ideal for generalization

Reply:
> cheesegull is really not a great protocol lol

not great but good enough, no reason to swap it out if all mirrors support it rn

Reply:
perhaps best to abstract this into functions, like handle_cheesegull(), handle_osudirect(), etc.

but for another pr

Reply:
we can discuss that tomorrow on the discord then I'll do a PR after this is merged

Reply:
> should still keep this case as it's a significant simplification if mirrors want to support serialization to osu!direct format

mino supports osu!direct format since it's creation with the "raw" query

i'm still open to rather return a response that uses the osu!api v2 format instead of cheesegull because, seriously why was that even a thing?

https://catboy.best/api/search?raw=1

Reply:
when bpy was made, all existing mirrors used exclusively cheesegull. everybody used ripple which only supported that format, so it just continued… i think api v2 format is definitely better, but it’s a breaking change that will likely stop a lot of mirrors being compatible which seems like the opposite goal of this pr

Reply:
if we don't make this breaking change it won't give mirrors a reason to switch away. I would welcome this change and i'm sure i'm not the only one who sees that, even speaking from a mirror perspective. We would be more of a mirror by using the same responses as osu!api v2. It would also profit other developers outside bancho.py

Reply:
> if we don't make this breaking change it won't give mirrors a reason to switch away
as much as this is true, we have to consider who is effected by such breaking change. we should support it but not enforce it.

the point of this pr to generalise mirror usage, and introducing a change supported by 2 mirrors (yours, and mine) is really against that. i’d rather use this pr as a “hey mirror developers, pls support api v2 going forward” and THEN we can talk about requiring it’s usage. this is too much destruction.

Reply:
yeah i agree we can and should get mirror devs to support it. i just don’t want anything which enforced it’s use until the majority/ all support it

Reply:
tbh I don't see a disadvantage of cheesegull. It's bad design but it is what established. Not only do private servers rely on cheesegull but also many discord bots, websites etc.

Reply:
> i still think this change is necessary in near future to have it more generalized

How would establishing a new standard help with generalization?

Reply:
> 

+

Reply:
> tbh I don't see a disadvantage of cheesegull. It's bad design but it is what established. Not only do private servers rely on cheesegull but also many discord bots, websites etc.

+1

Reply:
> tbh I don't see a disadvantage of cheesegull. It's bad design but it is what established. Not only do private servers rely on cheesegull but also many discord bots, websites etc.

I politely disagree. The main reason i dislike cheesegull for example is of how unfriendly it is for developers for being supposed to be a mirror for osu. People have to write a completely different parser if they want to add a mirror into their bot as a fallback to osu or vice versa. Also Cheesegull may give less beatmap information than just a simple osu!api v2 return. You agreeing that it's a bad design confuses me when you also say that you don't see any disadvantages.
Giving it "a new standard" also sounds weird if we think about the fact that we following the original standard should be used in the first place. Like i said, if noone changes something, then it never will become a standard and it gets harder to migrate away from it.

Reply:
using the original structure is simply "the standard" we should have been using instead of cheesegull. It isn't new, it's been already there since ever and there is no reason to not use it since we pratically don't *need* Cheesegull.

Reply:
Well reasons not not switch off have been named in this convo. Yes Cheesegull has flaws and there should be a different standard but that's how standards work, it's hard to just change them.